gonzo/clouds/compute.py

```diff
@@ -278,6 +286,65 @@ def create_volume(self, instance, vol_name,
             ex_volume_type=vol_type,
         )
 
+    def create_if_not_exist_security_group(self, group_name, subnet_id=None):
+        if subnet_id is not None:
+            subnet = self.compute_session.ex_list_subnets([subnet_id])[0]
+            subnet_vpc = subnet.extra['vpc_id']
+        else:
+            subnet_vpc = None
+
+        try:
+            desc = "Rules for {}".format(group_name)
+            self.compute_session.ex_create_security_group(
+                group_name,
+                desc,
+                subnet_vpc,
+            )
+        except Exception as exc:  # libcloud doesn't raise anything better
+            if not ("exists" in str(exc)):
+                raise
+
+    def get_subnet(self, subnet_id):
+        for subnet in self.compute_session.ex_list_subnets():
+            if subnet.id == subnet_id:
+                return subnet
+        raise LookupError("Subnet containting subnet: {} not found".format(
+            subnet_id))
+
+    def get_security_group(self, group_name, subnet_id=None):
+        filters = None
+        if subnet_id:
+            filters = {'vpc-id': self.get_vpc_for_subnet(subnet_id)}
+
+        sec_groups = self.compute_session.ex_get_security_groups(
+            filters=filters)
+
+        for sec_group in sec_groups:
+            if sec_group.name == group_name:
+                return sec_group
+
+    def get_vpc_for_subnet(self, subnet_id):
+        try:
+            subnet = self.compute_session.ex_list_subnets([subnet_id])
```

Comment:
how come you prefer this to `get_subnet`? you could handle both the `LookUpError` and `IndexError` then.